services/importer/lib/importer/base_connector.rb

```diff
@@ -0,0 +1,326 @@
+        # (this is to allow its use in parameters such as sql_query)
+        conn_params.each do |key, value|
+          PARAM_SEP_TOKENS.each do |token, symbol|
+            value = value.gsub(token, symbol)
+          end
+          conn_params[key] = value
+        end
+
+        # Extract columns of the result table from the parameters
+        # Column definitions are expected in SQL syntax, e.g.
+        #     columns=id int, name text
+        @columns = conn_params.delete 'columns'
+        @columns = @columns.split(',').map(&:strip) if @columns
+        @params = Cartodb::config[:fdw].merge(conn_params)
+      end
+
+      def validate_params!
+        errors = []
+        if @params['dsn'].blank? &&
+           (@params['driver'].blank? || @params['database'].blank? || @params['table'].blank?)
+          errors << "Missing required parameters"
+        end
+        if @params['channel'].blank? || @params['channel'].casecmp(channel_name) != 0
+          print "IMPORTER: channel name #{channel_name}\n"
+          print "IMPORTER: compare #{@params['channel'].casecmp(channel_name)}\n"
+          raise InvalidChannelError.new(@params['channel'], @user)
+        end
+        #errors << "Missing columns definition" unless @columns.present?
+        invalid_params = @params.keys - accepted_parameters
+        errors << "Invalid parameters: #{invalid_params * ', '}" if invalid_params.present?
+        raise InvalidParametersError.new(errors * "\n") if errors.present?
+      end
+
+      def connector_name
+        @params['dsn'] || @params['driver']
+      end
+
+      def server_name
+        v = "connector_#{connector_name}_#{@unique_suffix}"
+        print "IMPORTER: server_name #{v}\n"
+        v
+      end
+
+      def foreign_table_name
+        v = "#{server_name}_#{@params['table']}"
+        print "IMPORTER: foreign_table_name #{v}\n"
+        v
+      end
+
+      def run_create_server
+        execute_as_superuser create_server_command
+      end
+
+      def create_server_command
+        options = server_params.map { |k, v| "#{k} '#{v}'" } * ",\n"
+        v = %{
+          CREATE SERVER #{server_name}
+            FOREIGN DATA WRAPPER #{@params['channel']}
+            OPTIONS (#{options});
+        }
+        print "IMPORTER: create_server_command #{v}\n"
+        v
+      end
+
+      def run_create_user_mapping
+        execute_as_superuser create_user_mapping_command
+      end
+
+      def create_user_mapping_command
+        v = %{
+          CREATE USER MAPPING FOR "#{@user.database_username}" SERVER #{server_name};
+        }
+        print "IMPORTER: create_user_mapping_command #{v}\n"
+        v
+      end
+
+      def run_create_foreign_table
+        execute_as_superuser create_foreign_table_command
+      end
+
+      def create_foreign_table_command
+        options = table_params.map { |k, v| "#{k} '#{v}'" } * ",\n"
+        print "IMPORTER: foreign table options: #{options}\n"
+        v = %{
```

Comment:
why don't use the functions we have? https://github.com/CartoDB/cartodb-postgresql/blob/master/scripts-available/CDB_ForeignTable.sql